src/kubectlUtils.ts

```diff
@@ -0,0 +1,107 @@
+import * as vscode from "vscode";
+import { Kubectl } from "./kubectl";
+
+export type Cluster = {
+    name: string;
+    context: string;
+    user: string;
+    active: boolean;
+};
+
+export type Namespace = {
+    name: string;
+    active: boolean;
+};
+
+async function getKubeconfig(kubectl: Kubectl): Promise<any> {
+    const shellResult = await kubectl.invokeAsync("config view -o json");
+    if (shellResult.code !== 0) {
+        vscode.window.showErrorMessage(shellResult.stderr);
+        return null;
+    }
+    return JSON.parse(shellResult.stdout);
+}
+
+export async function getClusters(kubectl: Kubectl): Promise<Cluster[]> {
+    const kubectlConfig = await getKubeconfig(kubectl);
+    if (!kubectlConfig) {
+        return;

```

Comment:
Not keen on a bald `return;` in a function that is meant to return a value.  (I'm surprised the compiler doesn't complain about this!  We should consider turning it on in the linter if that supports it.)  If we are comfortable with returning `undefined` then let's make that explicit by writing `return undefined;`.  Let's also be clear in that case about when we use `null` vs `undefined` - the internal `getKubeconfig` function returns `null` for failure, rather than `undefined`.

Reply:
If the caller is not expected to handle `undefined/null` then consider returning an empty array... long term I'd like to get to all functions being clear about whether they are guaranteed to return a value, or only optionally (e.g. `Promise<Cluster[]>` vs `Promise<Cluster[] | null>`.  That might be a long road though!

Reply:
Agree.